modules/loader.js

```diff
@@ -846,90 +842,87 @@ __private.snapshotFinished = err => {
  * @todo Add description for the params
  */
 __private.loadBlocksFromNetwork = function(cb) {
-	let errorCount = 0;
+	// Number of failed attempts to load from peers.
+	let failedAttemptsToLoad = 0;
+	// If True, own node's db contains same number of blocks than the asked peer.
 	let loaded = false;
 
 	async.whilst(
-		() => !loaded && errorCount < 5,
-		next => {
-			self.getNetwork((err, network) => {
-				if (err) {
-					errorCount += 1;
-					return next();
-				}
-				const peer =
-					network.peers[Math.floor(Math.random() * network.peers.length)];
-				let lastBlock = modules.blocks.lastBlock.get();
-
-				function loadBlocks() {
-					__private.blocksToSync = peer.height;
-
-					modules.blocks.process.loadBlocksFromPeer(
-						peer,
-						(err, lastValidBlock) => {
+		() => !loaded && failedAttemptsToLoad < 5,
+		whilstCb => {
+			async.waterfall(
+				[
+					waterfallCb => {
```

Comment:
Suggest tp use function names to make it more readable as the steps.

Reply:
I've scanned the code for examples and I realise there is a mix of all sort of cases. I'll open an issue to make it uniform.

Reply:
Same as above, the code you are refactoring, should be fixed right in same PR.

Reply:
Same as above but even more clear https://github.com/LiskHQ/lisk/commit/24e363b6892e98147cefd7e8dae36cf1dce4ded6

Reply:
I mean it is a **recent** commit of yours using the wrong approach you just comment about. The criteria is clearly not clear.
Please, keep your review in the scope of the issue.

Reply:
The scope of any review includes to check if the proper conventions are practices are followed or not.

Reply:
Please, point me to the proper convention of waterfall.

Reply:
The error object convention already refactored, just use the function names in the async steps.